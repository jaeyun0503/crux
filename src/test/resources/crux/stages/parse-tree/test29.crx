// test int comparison

fn blt(a: int, b: int) -> bool {
  return a < b;
}

fn main() {
  printBool(blt(1,2));
  println();
  printBool(blt(2,1));
  println();
  printBool(blt(1,1));
}
