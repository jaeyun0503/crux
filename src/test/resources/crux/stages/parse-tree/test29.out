program
  declList
    decl
      functionDefn
        paramList
          param
            type
          param
            type
        type
        stmtBlock
          stmtList
            stmt
              returnStmt
                expr0
                  expr1
                    expr2
                      expr3
                        designator
                  op0
                  expr1
                    expr2
                      expr3
                        designator
    decl
      functionDefn
        paramList
        stmtBlock
          stmtList
            stmt
              callStmt
                callExpr
                  exprList
                    expr0
                      expr1
                        expr2
                          expr3
                            callExpr
                              exprList
                                expr0
                                  expr1
                                    expr2
                                      expr3
                                        literal
                                expr0
                                  expr1
                                    expr2
                                      expr3
                                        literal
            stmt
              callStmt
                callExpr
                  exprList
            stmt
              callStmt
                callExpr
                  exprList
                    expr0
                      expr1
                        expr2
                          expr3
                            callExpr
                              exprList
                                expr0
                                  expr1
                                    expr2
                                      expr3
                                        literal
                                expr0
                                  expr1
                                    expr2
                                      expr3
                                        literal
            stmt
              callStmt
                callExpr
                  exprList
            stmt
              callStmt
                callExpr
                  exprList
                    expr0
                      expr1
                        expr2
                          expr3
                            callExpr
                              exprList
                                expr0
                                  expr1
                                    expr2
                                      expr3
                                        literal
                                expr0
                                  expr1
                                    expr2
                                      expr3
                                        literal
